networking_aci/plugins/ml2/drivers/mech_aci/common.py

```diff
@@ -135,11 +137,94 @@ def get_hosts_on_network(self, context, network_id, level=None, with_segment=Fal
                 hosts.add(host)
         return hosts
 
+    def get_hosts_on_physnet(self, context, physical_network, level=None, with_segment=False, with_segmentation=False):
+        """Get all binding hosts (from host or binding_profile) present on a network"""
+        fields = [ml2_models.PortBinding.host, ml2_models.PortBinding.profile]
+        if with_segment:
+            fields.append(ml2_models.PortBindingLevel.segment_id)
+        if with_segmentation:
+            fields.append(segment_models.NetworkSegment.segmentation_id)
+        query = context.session.query(*fields)
+        query = query.join(ml2_models.PortBindingLevel,
+                           ml2_models.PortBinding.port_id == ml2_models.PortBindingLevel.port_id)
+        query = query.join(segment_models.NetworkSegment,
+                           ml2_models.PortBindingLevel.segment_id == segment_models.NetworkSegment.id)
+        query = query.filter(segment_models.NetworkSegment.physical_network == physical_network)
+        if level is not None:
+            query = query.filter(ml2_models.PortBindingLevel.level == level)
+
+        hosts = set()
+        for entry in query.all():
+            host = get_host_from_profile(entry.profile, entry.host)
+            if with_segment or with_segmentation:
+                row = [host]
+                if with_segment:
+                    row.append(entry.segment_id)
+                if with_segmentation:
+                    row.append(entry.segmentation_id)
+                hosts.add(tuple(row))
+            else:
+                hosts.add(host)
+        return hosts
```

Comment:
Can we comment how the return object looks here. When I initially saw the retun of this being consumed, it was not clear to me what is actually returned.

Reply:
Added docstring to method explaining its return behaviour.